api/src/pcapi/core/users/models.py

```diff
@@ -950,6 +950,10 @@ class GdprUserDataExtract(PcObject, Base):
     authorUserId: int = sa.Column(sa.BigInteger, sa.ForeignKey("user.id"), nullable=False)
     authorUser: orm.Mapped[User] = orm.relationship(User, foreign_keys=[authorUserId])
 
-    @property
+    @hybrid_property
     def expirationDate(self) -> datetime:
         return self.dateCreated + timedelta(days=7)
+
+    @expirationDate.expression  # type: ignore [no-redef]
+    def expirationDate(cls):  # pylint: disable=no-self-argument
+        return cls.dateCreated + timedelta(days=7)
```

Comment:
Question : timedelta marche-t-il bien dans une expression sql ?

Reply:
d'après ma compréhension oui : https://stackoverflow.com/questions/72693631/sqlalchemy-set-a-field-to-1-hour-in-the-future-using-func-now

sur l'exemple il utilise sa.func.now() + timedelta(hours=1)

Je ne sais pas comment vérifier cela par contre, j'imagine que cela fonctionne car lorsque je créer un ligne dans ma table gdpr, elle est bien filtré dans le front via l'expiration date grace à : 
`.filter(users_models.GdprUserDataExtract.expirationDate >= sa.func.now())`

test avec id 100 avec une vieille dateCreated: 
<img width="715" alt="Capture d’écran 2024-06-04 à 17 34 52" src="https://github.com/pass-culture/pass-culture-main/assets/114911174/2e0cc194-7b9d-431f-9e70-32f442f574dc">

pas d'id 100 dans le front, cela a bien été filtré:
<img width="364" alt="Capture d’écran 2024-06-04 à 17 35 35" src="https://github.com/pass-culture/pass-culture-main/assets/114911174/8617cea7-08bd-4e65-bd95-22b30ccbdbe2">


N'hésite pas à m'expliquer si je n'ai pas compris quelque chose à ce sujet